pkg/game/board.go

```diff
@@ -399,21 +415,26 @@ func (board *board) ScoreRoadCompletion(tile elements.PlacedTile, road feature.F
 	}
 
 	// check road in "left" direction
-	roadFinishedResult, scoreResult, meeplesResult, loopResult, loopSide = board.CheckRoadInDirection(leftSide, tile)
+	roadFinishedResult, scoreResult, meeplesResult, loopResult, loopSide, finishedPosLeft := board.CheckRoadInDirection(leftSide, tile)
 	score += scoreResult
 	roadFinished = roadFinished && roadFinishedResult
 	meeples = append(meeples, meeplesResult...)
 
 	// check road in "right" direction
 	if !loopResult && rightSide != side.NoSide {
-		roadFinishedResult, scoreResult, meeplesResult, _, _ = board.CheckRoadInDirection(rightSide, tile)
+		roadFinishedResult, scoreResult, meeplesResult, _, _, finishedPosRight := board.CheckRoadInDirection(rightSide, tile)
 		score += scoreResult
 		roadFinished = roadFinished && roadFinishedResult
 		meeples = append(meeples, meeplesResult...)
+
+		// check if loop is on different tile
```

Comment:
I tried improving the explanation, what do you think?
```suggestion
		// Decrement the score to prevent counting the tile twice
		// when its road features (two different ones) are both the start
		// and the end of the road (as is the case for crossroads).
		//
		// Note that this is a different scenario than a literal loop (i.e. an actual circle
		// with no end or beginning) where we don't fall into this `if` branch at all
		// due to `loopResult` being `true`.
```